### DISCUSSION TITLE
🚀 Any chance to get it working on MacOS and Windows?

### DISCUSSION THREAD

**Discussion by gerardbosch** [erikw/restic-automatic-backup-scheduler#47] [Q&A] [answered]
Hi! I'm wondering how hard would be to make this work on MacOS as well :rocket:.

I don't know much about that, and I think Mac does not use systemd but another init system. Anyway, I'm so happy with the triad "restic / Backblaze / this script" that I'd love :heart: to have it working on MacOS, so we can have a single "universal" backup solution for both Linux and MacOS.

I guess it should be feasible and probably not so complicated in the case that Mac init system offers similar capabilities than systemd.

What do you all think? Anyone has ideas or prior background?

Cheers!

P.S. My use case is to backup homedir only, but I don't think this is much of important.

**Comment by erikw** [comment, marked as answer]
Hello,

* Mac
  * I'm using the simple cron approach described in the README as when I got my Mac I was too lazy to make it more advanced. I've run the cron setup several years and it works good.
  * The better way would probably be to use `LaunchAgent` for this though as documented [here](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/ScheduledJobs.html#//apple_ref/doc/uid/10000172i-CH1-SW2). It does not seem hard to do.
* Windows
  * I currently don't have any Windows systems but up until last year I did. I hacked something togheter that worked quite well using the built-in `ScheduledTasks`. See here: https://github.com/erikw/dotfiles/tree/personal/bin/restic-windows


To answer your question, to get it to work on Mac/win: it already does!

Most effort would be to document this nicely in the repo, and figure out how to structure and separate this. Should the Makefile have different targets like `make install-systemd`, `make install-mac, make install-windows`, or create some other solution than the Makefile?

I could get this done, not right now, but a bit later when I have more time.

Are you on Mac & Windows as well? In that case I could need someone help in testing it out!

**Reply by gerardbosch** [reply]
Oh! That's coo! I forgot that cron can be used on mac :facepalm: 
I'm not on macOS right now, so I cannot test any further on that platform.

Thanks again for providing this solution :smiley:

**Reply by erikw** [reply]
For those who come in here to look, both macOS and Windows are now supported!

See the updated [README.md](https://github.com/erikw/restic-systemd-automatic-backup) for instructions for each OS type.
